content/en/docs/how-tos/adding-a-new-secret-to-ci.md

```diff
@@ -4,9 +4,66 @@ date: 2020-12-21T10:08:01-04:00
 draft: false
 ---
 
+If a job needs to access sensitive information, we can seal it into a [Kubernete secret](https://kubernetes.io/docs/concepts/configuration/secret/) and mount it to the pod which runs the job. There are several OpenShift [clusters in CI build farm](/docs/getting-started/useful-links/#clusters). A user can maintain the secrets in a namespace that the user has access to on `api.ci`. The CI `secret-mirroring` tool can be used to mirror the secrets to other namespaces on all clusters in the build farm. 
+
+## Add a new secret
+
 Choose a new namespace, and create a pull request to add a folder with the same name in [release/core-services](https://github.com/openshift/release/tree/master/core-services). The folder should contain the manifests for the namespace and the RBACs regarding to the namespace. If the desired namespace exists already, ask the owners of the namespace if it can be shared. After the pull request is merged, the manifests in the folder will be applied automatically to `api.ci`.
 
 Log into `api.ci` and create the secret in the namespace chosen above.
 
 Use the secret as the source in [the mapping file](https://github.com/openshift/release/tree/master/core-services/secret-mirroring).
 The secret and any modification on it afterwards will be populated to the targeting namespace on [all clusters](/docs/getting-started/useful-links/#clusters) in the build farm.
+
+```yaml
+- from:
+    namespace: some-namespace
+    name: source-some-secret
+  to:
+    namespace: test-credentials # the suggested namespace holding all secrets used in a step
+    name: target-some-secret # ensure the secret name unique cross all teams in our organization
+```
+
+## Use a secret in a job
+
+* The most common case is to use secrets in a [step](/docs/architecture/step-registry/#step). In this case, we suggest the user to mirror secrets to `test-credentials` namespace. The pod which runs the step can access the secrets defined in `credentials` stanza of the step definition. see [injecting-custom-credentials](https://docs.ci.openshift.org/docs/architecture/step-registry/#injecting-custom-credentials) for details.
+
+* For a job which is generated from `ci-operator`'s config and does not use steps, we have to use `ci` as the targeting namespace in secret-mirroring. And then mount the secrets via `secrets` stanza in the ci-operator's config, e.g.,
```

Comment:
I would rather suggest users always use `steps` - we can omit this section

Reply:
I think we should keep this information (because we *do* have handcrafted jobs), but clearly make this an advanced, not preferred option. Probably under a separate header.

Reply:
Put them into a new section.
See if the message (do not use them unless there is no other options) is delivered clearly.